blockly/blocks/arduino/io.js

```diff
@@ -199,15 +199,23 @@ Blockly.Blocks['io_tone'] = {
         .appendField(new Blockly.FieldDropdown(
             Blockly.Arduino.Boards.selected.digitalPins), "TONEPIN");
     this.appendValueInput("FREQUENCY")
-        .setCheck("Number")
-        .appendField("at frequency (in range 31 - 65535)");
+        .setCheck(Blockly.Types.NUMBER.compatibles())
+        .appendField("at frequency");
     this.setInputsInline(true);
     this.setPreviousStatement(true);
     this.setNextStatement(true);
     this.setColour(Blockly.Blocks.io.HUE);
-    this.setTooltip("Set's tone on pin to specified frequency");
+    this.setTooltip("Sets tone on pin to specified frequency within range 31 - 65535");
     this.setHelpUrl('https://www.arduino.cc/en/Reference/tone');
   },
+  onchange: function() {
+    var freq = Blockly.Arduino.valueToCode(this, "FREQUENCY", Blockly.Arduino.ORDER_ATOMIC)
+    if (freq < 31 || freq > 65535) {
+      this.setWarningText("Frequency must be in range 31 - 65535");
+    } else {
+      this.setWarningText(null);
```

Comment:
If we add the second argument to `setWarningText` the message gets associated with that id, and this call clearing could just target that and would not inadvertently remove any other warnings placed on the block by a different piece of code.